src/Runtime/Values.hs

```diff
@@ -81,3 +81,14 @@ instance Show Val where
   show (Pv _ e)       = show e
   show (Err s)        = "ERR: " ++ s
   show (Deferred)     = "Deferred"
+
+showValType :: Val -> String
```

Comment:
As mentioned earlier, the user input should be type checked. That obviates this function, since we can just show the type directly. The issue with having a second function for showing types is that it duplicates functionality and we have to keep the two functions in sync.